### DISCUSSION TITLE
Registration button doesn't work

### DISCUSSION THREAD

**Discussion by Pyrrha** [firefly-iii/firefly-iii#4497] [Firefly III] [answered]
**Bug description**
I am running Firefly III in latest version (from Docker jc5x/firefly-iii:latest), and I can't validate registeration on the /register form.

**Steps to reproduce**

- Deploy it on a Docker Swarm infrastructure
- Access to Firefly
- Fill registration form

**Extra info**

Can't access on /debug page.

**Bonus points**

- I searched and nobody reported this bug before
- I've asked some other people to access and try to register from different computers, IP, and browser.
- I have added a stack trace from my log files 
```
10.0.25.38 - - [12/Mar/2021:13:29:38 +0100] "GET /login HTTP/1.1" 302 2158 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.3 Safari/605.1.15"
10.0.25.38 - - [12/Mar/2021:13:29:38 +0100] "GET /register HTTP/1.1" 200 4031 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.3 Safari/605.1.15"
10.0.25.38 - - [12/Mar/2021:13:29:49 +0100] "GET /debug HTTP/1.1" 302 2143 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.3 Safari/605.1.15"
[2021-03-12T13:29:49.451871+01:00] local.INFO: AUDIT: Show login form (1.1). (10.0.25.38 -> GET:http://compta.dietz.dev/login) [] []
```

It seems that the button isn't working on this version, but works fine when using docker-compose.

**Comment by JC5** [comment]
How many nodes in the swarm?

**Reply by JC5** [reply]
The problem is that if you use a swarm with multiple nodes and you use the default storage mechanisms like `file` for session management the nodes will not be able to keep sessions in sync and will keep giving you different sessions. Basically, you have a 1/n chance of getting registered where n is the number of nodes.

You must switch to a shared storage driver like redis for the cache and the session driver.

**Reply by Pyrrha** [reply]
Hello,

Currently 2, but only 1 replicat of firefly and database. My aim is only to mount it on Swarm, but not replicate the site so far.

However I’ll try your advice, but I’m not sure it’ll change anything since it’s a single instance.

Thanks

**Reply by Pyrrha** [reply]
Hello 👋 

Just for you to know, it was Traefik that was an issue. I added it as proxy.

Thanks 🙏

**Reply by Chive** [reply]
Hey @Pyrrha I'm running into the same issue with traefik. Could you share how you have resolved it? Many thanks!

**Comment by Pyrrha** [comment, marked as answer]
@Chive the problem for me was the proxies configuration in Firefly III. I added `**` as suggested in comments, but you can specify your Traefik instance to be, I hope, more secure about data transmission.
